### DISCUSSION TITLE
How to obtain the carry-over effect from each marketing channel

### DISCUSSION THREAD

**Discussion by yongqiangzhangzazzle** [google/meridian#281] [Q&A] [answered]
Since the ads has the carry-over (or lagging) effect, how does Meridian MMM models the carry-over effect and in the results how to get the carry-over effect for each marketing channel? For instance, I'd like to know how many days the effect of the Google PLA ads last? Thanks!

**Comment by gauravsab1** [comment, marked as answer]
You can see how Meridian models the carryover effect [here](https://developers.google.com/meridian/docs/basics/media-saturation-lagging#adstock_function). It is modeled using the Adstock function. At a high-level, a media channel has a composite "impression stock" at a given time, say, *t*. And this stock is a weighted-average of impressions from time *t* going back *L* periods (where *L* is the chosen [max_lag](https://developers.google.com/meridian/docs/advanced-modeling/set-max-lag-parameter) parameter). The default value for this max_lag parameter is 8 weeks. 

Within the Adstock function, the effect of media in previous periods decays at rate "alpha" (the geometric decay rate, as described [here](https://developers.google.com/meridian/docs/basics/model-spec)). Media from periods further back in time than *L* periods do not, by definition, have an effect on the KPI/outcome in the current period. 

You can use functions such as the one [here](https://developers.google.com/meridian/reference/api/meridian/model/adstock_hill/AdstockTransformer?hl=en) to compute the Adstock and Hill functions.
